from aiogram import Router, F
from aiogram.types import Message
from bot.keyboards import connection_kb
import vpn_client
import logging

router = Router()


@router.message(F.text == "‚ö°Ô∏è –ü–æ–¥–∫–ª—é—á–∏—Ç—å—Å—è")
async def connect_button(message: Message):
    user_id = message.from_user.id
    if vpn_client.xui.session_string:
        # –£–∂–µ –∑–∞–ª–æ–≥–∏–Ω–µ–Ω, –º–æ–∂–Ω–æ –ø—Ä–æ–¥–æ–ª–∂–∞—Ç—å —Ä–∞–±–æ—Ç—É
        logging.info("Already logged in, using existing session")
        res = "‚úÖ –£–∂–µ –∞–≤—Ç–æ—Ä–∏–∑–æ–≤–∞–Ω—ã –Ω–∞ –ø–∞–Ω–µ–ª–∏."
    else:
        try:
            res = await vpn_client.get_client_key(user_id)
            logging.info(f"Login result: {res}")
            if res == "üî• Successful login":
                res = "‚úÖ –£—Å–ø–µ—à–Ω–æ –∞–≤—Ç–æ—Ä–∏–∑–æ–≤–∞–ª–∏—Å—å –Ω–∞ –ø–∞–Ω–µ–ª–∏."
            else:
                res = "‚ùå –ù–µ —É–¥–∞–ª–æ—Å—å –∞–≤—Ç–æ—Ä–∏–∑–æ–≤–∞—Ç—å—Å—è."
                
            res = "‚úÖ –£—Å–ø–µ—à–Ω–æ –∞–≤—Ç–æ—Ä–∏–∑–æ–≤–∞–ª–∏—Å—å –Ω–∞ –ø–∞–Ω–µ–ª–∏."
        except Exception as e:
            logging.error(f"Exception in connect_button: {e}")
            res = "‚ùå –ù–µ —É–¥–∞–ª–æ—Å—å –∞–≤—Ç–æ—Ä–∏–∑–æ–≤–∞—Ç—å—Å—è."
    await message.answer(res)
